File: tests/nodejs/cases/case_155_thread_handoff_late_06.js
/**
 * Task case 155: thread_handoff_late_06 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case155ThreadHandoffLate06(input) {
  const taskName = 'thread_handoff_late_06';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: async microtask captures payload beyond return edge.
  queueMicrotask(() => retainedAudit.push(payload));
  return 'ok';
}

module.exports = {
  case155ThreadHandoffLate06,
};

File: tests/go/cases/case_155_thread_handoff_late_06.go
package escape_tests

var retainedCase155 = []map[string]string{}

func Case155ThreadHandoffLate06(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "thread_handoff_late_06",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "thread_handoff_late_06:" + raw,
	}
	// ESCAPE: goroutine captures payload and appends to retained sink.
	go func(p map[string]string) {
		retainedCase155 = append(retainedCase155, p)
	}(payload)
	return "ok"
}
